### PR TITLE
Add client-side validation to teacher and principal application

### PR DESCRIPTION
**Pull request by hacodeorg**
# Description
Enforce the following constraints from the client side
**In Teacher Application**
- Class section minutes per day must be positive number from 1-480 inclusive
- Class section days per week must be positive number between 1-7 inclusive
- Class section weeks per year must be positive number from 1-52 inclusive
- School Zip Code must be a number
Screen Shot 2019-10-15 at 3 01 14 PM
Screen Shot 2019-10-15 at 2 57 12 PM

**In Principal approval application**
- Student enrollment must be positive integer
- Avoid turning empty demographics questions into "N/A" when submitting. (They are currently turned into "N/A" because we call `parseFloat` on an empty value, and then transform the result into a string.)
- School zip code should be validated

Screen Shot 2019-10-15 at 3 14 12 PM
Screen Shot 2019-10-15 at 3 14 33 PM

## Links
- [jira](https://codedotorg.atlassian.net/browse/PLC-231)

### CODE REVIEW

File: apps/src/code-studio/pd/application/teacher/ChooseYourProgram.jsx [hunk 1]
```diff
@@ -207,4 +207,36 @@ export default class ChooseYourProgram extends LabeledFormComponent {
...
+      formatErrors.csHowManyDaysPerWeek =
+        'Class section days per week must be positive number between 1-7 inclusive';
+    }
+
+    if (
+      data.csHowManyWeeksPerYear &&
+      (data.csHowManyWeeksPerYear < 1 || data.csHowManyWeeksPerYear > 52)
+    ) {
+      formatErrors.csHowManyWeeksPerYear =
+        'Class section weeks per year must be positive number from 1-52 inclusive';
```
Comment (agealy) [inline apps/src/code-studio/pd/application/teacher/ChooseYourProgram.jsx]: Is this error text user-facing?

Comment (hacodeorg) [reply apps/src/code-studio/pd/application/teacher/ChooseYourProgram.jsx]: Yes. Do you want to change it?

IMO, a better user experience could be that we show the expected values before users try to answer a question. That would eliminate a lot of confusion and mistakes. However, that is another discussion.

Comment (agealy) [reply apps/src/code-studio/pd/application/teacher/ChooseYourProgram.jsx]: Agreed on UX—primary motivation for now is to keep clean up the data. Let's tweak the text to: "Class section weeks per year must be between 1 and 52". Just going for something slightly more friendly. I'll leave a couple other comments with similar requests

File: apps/src/code-studio/pd/application/teacher/ChooseYourProgram.jsx [hunk 2]
```diff
@@ -207,4 +207,36 @@ export default class ChooseYourProgram extends LabeledFormComponent {
...
+      formatErrors.csHowManyMinutes =
+        'Class section minutes per day must be positive number from 1-480 inclusive';
+    }
+
+    if (
+      data.csHowManyDaysPerWeek &&
+      (data.csHowManyDaysPerWeek < 1 || data.csHowManyDaysPerWeek > 7)
+    ) {
+      formatErrors.csHowManyDaysPerWeek =
+        'Class section days per week must be positive number between 1-7 inclusive';
```
Comment (agealy) [inline apps/src/code-studio/pd/application/teacher/ChooseYourProgram.jsx]: Suggested error text: "Class section days per week must be between 1 and 7"

File: apps/src/code-studio/pd/application/teacher/ChooseYourProgram.jsx [hunk 3]
```diff
@@ -207,4 +207,36 @@ export default class ChooseYourProgram extends LabeledFormComponent {
...
+   * @override
+   */
+  static getErrorMessages(data) {
+    let formatErrors = {};
+    if (
+      data.csHowManyMinutes &&
+      (data.csHowManyMinutes < 1 || data.csHowManyMinutes > 480)
+    ) {
+      formatErrors.csHowManyMinutes =
+        'Class section minutes per day must be positive number from 1-480 inclusive';
```
Comment (agealy) [inline apps/src/code-studio/pd/application/teacher/ChooseYourProgram.jsx]: Suggested error text: "Class section minutes per day must be between 1 and 480"

File: apps/src/code-studio/pd/application/teacher/ChooseYourProgram.jsx (hunk 1 above)
Comment (hacodeorg) [reply apps/src/code-studio/pd/application/teacher/ChooseYourProgram.jsx]: All fixed.

**Review by agealy — APPROVED** [review]

**Review by islemaster — APPROVED** [review]
:+1: Very nice.

**Pull request merged**